src/app/projects/project-list/project-list-item.component.ts

```diff
@@ -1,29 +1,46 @@
-import { Component, EventEmitter, Input, OnChanges, Output } from '@angular/core';
+import {
+  Component,
+  EventEmitter,
+  Input,
+  OnChanges,
+  OnInit,
+  Output,
+} from '@angular/core';
 import { WharfProject } from 'src/app/models/main-project.model';
 import { NgChanges } from 'src/app/shared/util/ngchanges';
 import { LocalStorageProjectsService } from '../local-storage-projects.service';
 import { ProjectFavoriteClickEvent } from '../project-favorite/project-favorite-button.component';
 import { ProjectRefreshedEvent } from '../project-refresh';
 import { ProjectUtilsService } from '../project-utils.service';
+import { MenuItem } from 'primeng/api/menuitem';
 
 @Component({
   selector: 'wh-project-list-item',
   templateUrl: './project-list-item.component.html',
 })
-export class ProjectListItemComponent implements OnChanges {
+export class ProjectListItemComponent implements OnInit, OnChanges {
   @Input() project: WharfProject;
 
   @Output() refreshed = new EventEmitter<ProjectRefreshedEvent>();
   @Output() favoriteClick = new EventEmitter<ProjectFavoriteClickEvent>();
 
   isRefreshAnimationPlaying: boolean;
   providerHostname: string;
+  menuItems: MenuItem[];
+  noValidBuildStepsTooltip;
 
   constructor(
     public localStorageProjectsService: LocalStorageProjectsService,
     public projectUtilsService: ProjectUtilsService,
   ) { }
 
+  ngOnInit() {
+    this.menuItems = this.projectUtilsService.getActionsMenuItems(this.project);
+    if (!this.menuItems.length) {
+      this.noValidBuildStepsTooltip = 'There are no valid build steps for this project!';
```

Comment:
The correct term is "stages"

```suggestion
      this.noValidBuildStepsTooltip = 'There are no stages for this project!';
```

Reply:
Maybe rename the variable `this.noValidBuildStepsTooltip` as well then